src/main/java/teammates/ui/webapi/action/DeleteInstructorInCourseAction.java

```diff
@@ -63,12 +64,14 @@ private boolean hasAlternativeInstructor(String courseId, String instructorToDel
                             && !instr.getEmail().equals(instructorToDeleteEmail)
                             && instr.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
 
-            if (isAlternativeInstructor) {
+            boolean isAlternativeInstructorDisplayed =
+                    instr.isDisplayedToStudents() && !instr.getEmail().equals(instructorToDeleteEmail);
+
+            if (isAlternativeInstructor && isAlternativeInstructorDisplayed) {
```

Comment:
Splitting like this does not help. You still will only get `true` if all conditions are fulfilled, which is not what we want.

Reply:
My bad, it should be in separate for loops

Reply:
Is it resolved now?